### PR TITLE
Batchnorm tuning (re-entrant grid reduction)

### PR DESCRIPTION
**Pull request by naoyam**

### CODE REVIEW

**Comment by csarofeen** [comment]
Closing in favor of https://github.com/csarofeen/pytorch/pull/1698 @naoyam please double check there isn't anything in the tests you want to keep but I think we got everything. grdim tuning was a big part of the work in tuning outer reduction performance.

**Comment by naoyam** [comment]
Yes, everything is now in TOT.
